ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,394 @@
+The default is: `TLSv1 TLSv1.1 TLSv1.2`
+
+TLSv1 is enabled to allow old clients like:
+- [IE 8-10 / Win 7](https://www.ssllabs.com/ssltest/viewClient.html?name=IE&version=8-10&platform=Win%207&key=113)
+- [Java 7u25](https://www.ssllabs.com/ssltest/viewClient.html?name=Java&version=7u25&key=26)
+
+If you dont need to support this clients please remove TLSv1
+
+
+Please check the result of the configuration using `https://ssllabs.com/ssltest/analyze.html` or `https://testssl.sh`
+
+
+##### *ssl-dh-param:*
+
+Base64 string that contains Diffie-Hellman key to help with "Perfect Forward Secrecy"
+https://www.openssl.org/docs/manmaster/apps/dhparam.html
+https://wiki.mozilla.org/Security/Server_Side_TLS#DHE_handshake_and_dhparam
+http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_dhparam
+
+
+##### *ssl-session-cache:*
+
+Enables or disables the use of shared [SSL cache](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_session_cache) among worker processes.
+
+
+##### *ssl-session-cache-size:*
+
+Sets the size of the [SSL shared session cache](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_session_cache) between all worker processes.
+
+
+##### *ssl-session-tickets:*
+
+Enables or disables session resumption through [TLS session tickets](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_session_tickets)
+
+
+##### *ssl-session-timeout:*
+
+Sets the time during which a client may [reuse the session](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_session_timeout) parameters stored in a cache.
+
+
+##### *ssl-redirect:*
+
+Sets the global value of redirects (301) to HTTPS if the server has a TLS certificate (defined in an Ingress rule)
+Default is true
+
+
+##### *upstream-max-fails:*
+
+Sets the number of unsuccessful attempts to communicate with the [server](http://nginx.org/en/docs/http/ngx_http_upstream_module.html#upstream) that should happen in the duration set by the fail_timeout parameter to consider the server unavailable
+
+
+##### *upstream-fail-timeout:*
+
+Sets the time during which the specified number of unsuccessful attempts to communicate with the [server](http://nginx.org/en/docs/http/ngx_http_upstream_module.html#upstream) should happen to consider the server unavailable
+
+
+##### *use-proxy-protocol:*
+
+Enables or disables the use of the [PROXY protocol](https://www.nginx.com/resources/admin-guide/proxy-protocol/) to receive client connection (real IP address) information passed through proxy servers and load balancers such as HAproxy and Amazon Elastic Load Balancer (ELB).
+
+
+##### *use-gzip:*
+
+Enables or disables the use of the nginx module that compresses responses using the ["gzip" module](http://nginx.org/en/docs/http/ngx_http_gzip_module.html)
+
+
+##### *use-http2:*
+
+Enables or disables the [HTTP/2](http://nginx.org/en/docs/http/ngx_http_v2_module.html) support in secure connections 
```

Comment:
a great example would be loadbalancing a grpc server. LOTS of people have asked for it, just fyi.